modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpClient.java

```diff
@@ -46,20 +46,19 @@ HttpResponse send(HttpRequest request, int responseTimeout, boolean followRedire
       throws IOException, TimeoutException;
 
   /**
-   * Sends a HttpRequest without blocking the current thread. When a response is available or the request times out the provided
-   * {@link ResponseHandler} will be invoked. Be aware that the response body processing will be deferred so that the response can
+   * Sends a HttpRequest without blocking the current thread. When a response is available or the request times out the returned
+   * {@link CompletableFuture} will be completed. Be aware that the response body processing will be deferred so that the response can
    * be processed even when a large body is still being received. If the full response is needed right away then the provided
-   * {@link ResponseHandler} must execute in a different thread so that it does not block the {@link HttpClient} threads handling
-   * the response.
+   * {@link HttpResponse} must be read in a different thread so that it does not block the {@link HttpClient} threads handling the
```

Comment:
I'll run some tests because using CompletableFuture may have fixed it, but I may need to pass along your schedulers when consuming them. Good catch!